src/query/v2/requests.hpp

```diff
@@ -125,17 +335,23 @@ enum class StorageView { OLD = 0, NEW = 1 };
 
 struct ScanVerticesRequest {
   Hlc transaction_id;
-  size_t start_id;
-  std::optional<std::vector<std::string>> props_to_return;
+  VertexId start_id;
+  std::optional<std::vector<PropertyId>> props_to_return;
   std::optional<std::vector<std::string>> filter_expressions;
   std::optional<size_t> batch_limit;
   StorageView storage_view;
 };
 
+struct ScanResultRow {
+  Value vertex;
+  // empty() is no properties returned
+  std::map<PropertyId, Value> props;
+};
```

Comment:
I think we should merge as it is, there are other messages that will probably have to change until we have everything in place. Guessing what should be the final form of may not worth the effort we put in because they will possibly change.